presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -484,20 +493,59 @@ protected Scope visitCreateTableAsSelect(CreateTableAsSelect node, Optional<Scop
             // analyze the query that creates the table
             Scope queryScope = process(node.getQuery(), scope);
 
+            ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+
+            // analyze target table columns and column aliases
             if (node.getColumnAliases().isPresent()) {
                 validateColumnAliases(node.getColumnAliases().get(), queryScope.getRelationType().getVisibleFieldCount());
 
-                // analzie only column types in subquery if column alias exists
+                int aliasPosition = 0;
                 for (Field field : queryScope.getRelationType().getVisibleFields()) {
                     if (field.getType().equals(UNKNOWN)) {
                         throw semanticException(COLUMN_TYPE_UNKNOWN, node, "Column type is unknown at position %s", queryScope.getRelationType().indexOf(field) + 1);
                     }
+                    columns.add(new ColumnMetadata(node.getColumnAliases().get().get(aliasPosition).getValue(), field.getType()));
+                    aliasPosition++;
                 }
+
+                analysis.setCreateTableColumnAliases(node.getColumnAliases().get());
```

Comment:
`Analysis.getColumnAliases()` is not used anymore with this patch. Why do we need this call?